### PR TITLE
Section: Overview of game theoretic model

### PR DESCRIPTION
**Pull request by MichalisPanayides**

### CODE REVIEW

File: sections/model_overview/main.tex [hunk 1]
```diff
@@ -1 +1,168 @@
-\section{Overview of game theoretic model}
\ No newline at end of file
+\section{Overview of game theoretic model}
+
+The problem studied is a 3-player game theoretic scenario between two queueing 
+systems and a service that distributes individuals to them.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
The problem studied is a 3-player normal form game.

 The players are:
  
 \begin{itemize}
 
      \item The decision makers of two queueing 
systems;
      \item a service that distributes individuals to these two queueing systems.
 \end{itemize}
 
```

File: sections/model_overview/main.tex [hunk 2]
```diff
@@ -1 +1,168 @@
-\section{Overview of game theoretic model}
\ No newline at end of file
+\section{Overview of game theoretic model}
+
+The problem studied is a 3-player game theoretic scenario between two queueing 
+systems and a service that distributes individuals to them. 
+Each player in this game has its own objectives and cares about meeting its own 
+targets.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
This is a standard Normal form game~\cite{},  % TODO Cite a Game Theory text book (Maschler et al. would do)
in that each player in this game has their own objectives which they aim to optimise.
```

File: sections/model_overview/main.tex [hunk 3]
```diff
@@ -1 +1,168 @@
...
+More specifically, the queueing systems' objective is captured by an upper bound
+of the time that a fixed proportion of individuals need to spend in the system,
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
of the time that a fixed proportion of individuals  spend in the system, 
```

File: sections/model_overview/main.tex [hunk 4]
```diff
@@ -1 +1,168 @@
...
+\section{Overview of game theoretic model}
+
+The problem studied is a 3-player game theoretic scenario between two queueing 
+systems and a service that distributes individuals to them. 
+Each player in this game has its own objectives and cares about meeting its own 
+targets.
+More specifically, the queueing systems' objective is captured by an upper bound
+of the time that a fixed proportion of individuals need to spend in the system, 
+while the distributor aims to minimise the time that its individuals 
+are blocked.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
are blocked.   % TODO Depending on how this is described earlier, we might need a little more here (as to what we mean by "blocked")
```

File: sections/model_overview/main.tex [hunk 5]
```diff
@@ -1 +1,168 @@
...
+Each queueing system may then choose to block the individuals that arrive from 
+the distributor when the system reaches a certain capacity. 
+Thus, the set of strategies that they may choose to play in the game are defined 
+as the set of possible choices that the two queueing systems have. 
+In other words, each queueing system is given \( N_A \) and \( N_B \) number of 
+strategies where \( N_A \) and \( N_B \) are the total capacities of the two 
+queueing systems.
+The \textit{threshold} parameter is thus used to capture the chosen 
+strategy for each hospital where \( T_A \in \left[1, N_A \right] \) and 
+\( T_B \in \left[1, N_B \right] \)
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
The strategy sets for each queueing system is the set \(\{T \in \mathbb{N} \;|\; 1 \leq T \leq N \) and \( N_B \) 
where \(N\in\{N_A, N_B\}\) are the total capacities of the two queueing systems. We denote the chosen actions from the strategy set as \(T_A, T_B\) and call these \textit{threshold}s.
```

File: sections/model_overview/main.tex [hunk 6]
```diff
@@ -1 +1,168 @@
...
+Their choice of strategy will then rely solely on satisfying their own 
+objective, which is to make sure that the waiting time of a proportion of 
+individuals will be below the predefined target time.
+
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
individuals will be below the predefined target time.

This is shown diagrammatically in Figure~ref{fig:diagram_of_queueing_system}.

% TODO Draw diagram of queueing system
% This should include the parameters
% and clearly show the decision variable.
% Note: only do this for a single queueing system.

```

File: sections/model_overview/main.tex [hunk 7]
```diff
@@ -1 +1,168 @@
...
+\begin{equation}
+    P(W < \textit{ Target}) \geq \hat{P}
+\end{equation}
+
+where \(W\) is the waiting time of all individuals, \textit{Target} is the time
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
where \(W\) is the mean waiting time of all individuals, \textit{Target} is the time 
```

I'd not use `Target` in text at all (if you do it should not be in italics), just find some letter you haven't used yet and define that as the target.

File: sections/model_overview/main.tex [hunk 8]
```diff
@@ -1 +1,168 @@
...
+queueing systems.
+The \textit{threshold} parameter is thus used to capture the chosen 
+strategy for each hospital where \( T_A \in \left[1, N_A \right] \) and 
+\( T_B \in \left[1, N_B \right] \)
+Their choice of strategy will then rely solely on satisfying their own 
+objective, which is to make sure that the waiting time of a proportion of 
+individuals will be below the predefined target time.
+
+\begin{equation}
+    P(W < \textit{ Target}) \geq \hat{P}
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
    P(W < \text{ Target}) \geq \hat{P}
```

Italics within a math environment look like you're multiplying T by a by r by g by e by t.

File: sections/model_overview/main.tex [hunk 9]
```diff
@@ -1 +1,168 @@
...
+\end{equation}
+
+where \(W\) is the waiting time of all individuals, \textit{Target} is the time 
+target and \(\hat{P}\) is the percentage of individuals need to be within that 
+target. 
+There are numerous objective functions that can be used to capture this 
+behaviour. 
+The first one would be to use the threshold that maximises the probability that
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
For example one approach is to use the threshold that maximises the probability that 
```

File: sections/model_overview/main.tex [hunk 10]
```diff
@@ -1 +1,168 @@
...
+percentage goal.
+
+\begin{equation}
+    \arg \max_{T_i} \quad P(W_i < \textit{Target})
+\end{equation}
+
+Perhaps a more sophisticated objective function would be to get the proportion 
+of individuals as close to the percentage aim. 
+In other words, to find the threshold that minimises the difference between the 
+probability and the percentage goal (or maximise its negation).
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
A more sophisticated objective function would be to get the proportion 
of individuals as close to the percentage aim. 
In other words, to find the threshold that minimises the difference between the 
probability and the percentage goal (or maximise its negation).
```

File: sections/model_overview/main.tex [hunk 11]
```diff
@@ -1 +1,168 @@
...
+
+\begin{equation}\label{eq:obj-queueing-systems}
+    \arg \max_{T_i} \quad -\left( \hat{P} - P(W_i < \textit{Target}) \right)^2
+\end{equation}
+
+
+On the other side of the spectrum, the distributor has its own choices 
+to make and its own goals to satisfy.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
The third player, the distributor has their own choices 
to make and theirs own goals to satisfy.
```

File: sections/model_overview/main.tex [hunk 12]
```diff
@@ -1 +1,168 @@
...
+The set of strategies it has can be defined as the different ways that it can 
+distribute individuals to the two queueing system.
+Therefore, unlike the queueing systems, the distributor has a 
+continuous set of strategies that goes from \( 0\% - 100\% \). 
+For example, it may choose to send \(35\%\) of its individuals to the first 
+queueing system and \(65\%\) to the second one.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
The strategy set of the third player is the proportion \(0 \leq p \leq 1\) of individuals it sends to the first queueing system (the proportion \(1 - q\) is sent to the second queueing system).
```

File: sections/model_overview/main.tex [hunk 13]
```diff
@@ -1 +1,168 @@
...
+In addition, the distributor aims to minimise any potential blockages
+that may occur, given the pair of thresholds chosen by the two queueing systems.
+Thus, its objective is to minimise the blocked time of the individuals 
+that they send to the two queueing systems.
+Apart from the time being blocked, an additional aspect that may affect the 
+decision of the distribution service is the proportion of lost individuals.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]: We need to clarify above (somewhere near the diagram of the queueing system that individuals can be lost at all).

File: sections/model_overview/main.tex [hunk 14]
```diff
@@ -1 +1,168 @@
...
+between the two objectives.
+
+\begin{equation}\label{eq:obj-distributor}
+    \alpha P(L_A) + (1 - \alpha) B_A = 
+    \alpha P(L_B) + (1 - \alpha) B_B
+\end{equation}
+
+Here, the purpose of \(\alpha\) is to tune the ``importance'' of each objective,
+where high \(\alpha\) indicates a higher weight on the proportion of lost 
+individuals and smaller \(\alpha\) a higher weight on the time blocked.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
Here, \(\alpha\) represents the ``importance'' of each objective,
where high \(\alpha\) indicates a higher weight on the proportion of lost 
individuals and smaller \(\alpha\) a higher weight on the time blocked. 
```

File: sections/model_overview/main.tex [hunk 15]
```diff
@@ -1 +1,168 @@
...
+    \alpha P(L_B) + (1 - \alpha) B_B
+\end{equation}
+
+Here, the purpose of \(\alpha\) is to tune the ``importance'' of each objective,
+where high \(\alpha\) indicates a higher weight on the proportion of lost 
+individuals and smaller \(\alpha\) a higher weight on the time blocked. 
+
+
+Using equations \ref{eq:obj-queueing-systems} and \ref{eq:obj-distributor}, an
+imperfect information extensive form game may arise.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
Using equations \ref{eq:obj-queueing-systems} and \ref{eq:obj-distributor gives an
imperfect information extensive form game.
```

File: sections/model_overview/main.tex [hunk 16]
```diff
@@ -1 +1,168 @@
...
+where high \(\alpha\) indicates a higher weight on the proportion of lost 
+individuals and smaller \(\alpha\) a higher weight on the time blocked. 
+
+
+Using equations \ref{eq:obj-queueing-systems} and \ref{eq:obj-distributor}, an
+imperfect information extensive form game may arise. 
+An imperfect information game is defined as an extensive form game where some 
+of the information about the game state is hidden for at least one of the 
+players \cite{Berwanger2008}. In this study the state of the problem that is
+hidden is the threshold that each player chooses to play.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
hidden is the threshold that each of the first two players chooses to play.
```

File: sections/model_overview/main.tex [hunk 17]
```diff
@@ -1 +1,168 @@
...
+In other words, each queueing system chooses to play a strategy without the 
+knowing the other system's strategy.
+The distributor then, fully aware of the chosen strategies are, distributes
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
The distributor then, fully aware of the chosen threshold strategies, distributes 
```

File: sections/model_overview/main.tex [hunk 18]
```diff
@@ -1 +1,168 @@
...
+        \path[->] (D) edge node {}(D_d1);
+        \path[->] (D) edge node {}(D_d2);
+        \path(D_d1) edge [bend left] node {}(D_d2);
+    \end{tikzpicture}
+    \caption{Imperfect information Extensive Form Game between the distributor 
+    and the 2 queueing systems}
+    \label{fig:imperfect-info-game}
+\end{figure}
+
+Figure \ref{fig:imperfect-info-game} illustrates such a game works. The first
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
The first 
```

File: sections/model_overview/main.tex [hunk 19]
```diff
@@ -1 +1,168 @@
...
+imperfect information extensive form game may arise. 
+An imperfect information game is defined as an extensive form game where some 
+of the information about the game state is hidden for at least one of the 
+players \cite{Berwanger2008}. In this study the state of the problem that is
+hidden is the threshold that each player chooses to play.
+In other words, each queueing system chooses to play a strategy without the 
+knowing the other system's strategy.
+The distributor then, fully aware of the chosen strategies are, distributes 
+individuals among the two systems in order to minimise the time that its 
+individuals will be blocked.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
individuals will be blocked. Figure \ref{fig:imperfect-info-game} illustrates this. 
```

File: sections/model_overview/main.tex [hunk 20]
```diff
@@ -1 +1,168 @@
...
+    \begin{pmatrix}
+        U_{1,1}^B & U_{1,2}^B & \dots & U_{1,N_B}^B \\
+        U_{2,1}^B & U_{2,2}^B & \dots & U_{2,N_B}^B \\
+        \vdots & \vdots & \ddots & \vdots \\
+        U_{N_A,1}^B & U_{N_A,2}^B & \dots & U_{N_A,N_B}^B \\
+    \end{pmatrix}
+\end{equation}
+
+Based on the choice of strategy of these two players the distributor will then 
+make its own choice of the proportion of individuals to send to each system.
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
make their own choice of the proportion of individuals to send to each system.
```

**Review by drvinceknight — CHANGES_REQUESTED** [review]

File: sections/model_overview/main.tex (hunk 5 above)
Comment (MichalisPanayides) [reply sections/model_overview/main.tex]: That is looks definitely better.

File: sections/model_overview/main.tex (hunk 8 above)
Comment (MichalisPanayides) [reply sections/model_overview/main.tex]: Will keep in mind 👍

File: sections/model_overview/main.tex (hunk 7 above)
Comment (MichalisPanayides) [reply sections/model_overview/main.tex]: Hmmm🤔. Yeah, that makes sense. Will use `R` instead then 👍

File: sections/model_overview/main.tex (hunk 6 above)
Comment (MichalisPanayides) [reply sections/model_overview/main.tex]: By diagram do you mean something like the general queuing diagram shown below or the Markov chain one?

image

File: sections/model_overview/main.tex [hunk 21]
```diff
@@ -1 +1,249 @@
...
+        \draw (4.2, 0.525) -- (4.2, -0.75);
+        
+        % First line (horizontal) from lambda_1
+        \draw (4.2, 1.8) -- +(-169.5pt,0) node[left] {\( \lambda_1 \)};
+        % Second line (vertical) from lambda_1
+        \draw (4.2, 1.8) -- (4.2, 0.525);
+        % Arrow line to Queue 2
+        \draw[->] (4.2, 0.525) -- (4.8, 0.525);  
+    \end{tikzpicture}
+    \caption{Queueing Model}
```
Comment (drvinceknight) [inline sections/model_overview/main.tex]:
```suggestion
    \caption{A diagrammatic representation of the queueing model. 
    The threshold \(T\) only applies to arrivals from the first buffer. 
    If the second buffer is at that threshold only individuals of the 
    first type are accepted (at a rate \(\lambda_1\)) and individuals 
    of the second type (arriving at a rate \(\lambda_2\)) are held 
    in the blocked in the first buffer.}
```

Comment (drvinceknight) [reply sections/model_overview/main.tex]: Once that's in, let's merge this PR (and rebase #5)

**Pull request merged**